Mathlib/AlgebraicGeometry/Morphisms/Basic.lean

```diff
@@ -156,347 +233,248 @@ instance targetAffineLocally_respectsIso (P : AffineTargetMorphismProperty)
+    dsimp; infer_instance
+  rw [iff_of_openCover (P := P) (Scheme.openCoverOfIsIso.{0} (𝟙 Y))]
+  trans Q (pullback.snd f (𝟙 _))
+  · exact ⟨fun H => H PUnit.unit, fun H _ => H⟩
+  rw [← Category.comp_id (pullback.snd _ _), ← pullback.condition,
+    Q.cancel_left_of_respectsIso]
+#align algebraic_geometry.affine_target_morphism_property.is_local.affine_target_iff AlgebraicGeometry.HasAffineProperty.iff_of_isAffine
+
+instance (𝒰 : X.AffineOpenCover) (i) : IsAffine (𝒰.openCover.obj i) :=
+  inferInstanceAs (IsAffine (Spec (𝒰.obj i)))
+
+instance (priority := 900) : IsLocalAtTarget P := by
+  letI := isLocal_affineProperty P
+  apply IsLocalAtTarget.mk'
+  · rw [eq_targetAffineLocally P]
+    intro X Y f U H V
+    rw [Q.arrow_mk_iso_iff (morphismRestrictRestrict f _ _)]
+    exact H ⟨_, V.2.image_of_isOpenImmersion (Y.ofRestrict _)⟩
+  · rintro X Y f ι U hU H
+    let 𝒰 := Y.openCoverOfSuprEqTop U hU
+    apply of_openCover 𝒰.affineRefinement.openCover
+    rintro ⟨i, j⟩
+    have : P (𝒰.pullbackHom f i) := by
+      refine (P.arrow_mk_iso_iff
+        (morphismRestrictEq _ ?_ ≪≫ morphismRestrictOpensRange f (𝒰.map i))).mp (H i)
+      exact (opensRange_ιOpens _).symm
+    rw [← Q.cancel_left_of_respectsIso (𝒰.pullbackCoverAffineRefinementObj f _).inv,
+      𝒰.pullbackCoverAffineRefinementObj_inv_pullbackHom]
+    exact of_isPullback (.of_hasPullback _ _) this
+#align algebraic_geometry.affine_target_morphism_property.is_local.target_affine_locally_is_local AlgebraicGeometry.HasAffineProperty.instIsLocalAtTarget
+
+theorem pullback_fst_of_right (hP' : Q.StableUnderBaseChange)
+    {X Y S : Scheme} (f : X ⟶ S) (g : Y ⟶ S) [IsAffine S] (H : Q g) :
+    P (pullback.fst f g) := by
+  letI := isLocal_affineProperty P
+  rw [iff_of_openCover (P := P) X.affineCover]
   intro i
   let e := pullbackSymmetry _ _ ≪≫ pullbackRightPullbackFstIso f g (X.affineCover.map i)
-  have : e.hom ≫ pullback.fst _ _ = pullback.snd _ _ := by simp [e]
-  rw [← this, P.cancel_left_of_respectsIso]
-  apply hP'; assumption
-#align algebraic_geometry.affine_target_morphism_property.is_local.target_affine_locally_pullback_fst_of_right_of_stable_under_base_change AlgebraicGeometry.AffineTargetMorphismProperty.IsLocal.targetAffineLocally_pullback_fst_of_right_of_stableUnderBaseChange
-
-theorem IsLocal.stableUnderBaseChange {P : AffineTargetMorphismProperty} (hP : P.IsLocal)
-    (hP' : P.StableUnderBaseChange) : (targetAffineLocally P).StableUnderBaseChange :=
-  have := hP.1
+  have : e.hom ≫ pullback.fst _ _ = X.affineCover.pullbackHom (pullback.fst _ _) i := by
+    simp [e, Scheme.OpenCover.pullbackHom]
+  rw [← this, Q.cancel_left_of_respectsIso]
+  apply hP' _ _
+  exact H
+#align algebraic_geometry.affine_target_morphism_property.is_local.target_affine_locally_pullback_fst_of_right_of_stable_under_base_change AlgebraicGeometry.HasAffineProperty.pullback_fst_of_right
+
+theorem stableUnderBaseChange_mk (hP' : Q.StableUnderBaseChange) :
+    P.StableUnderBaseChange :=
```

Comment:
```suggestion
theorem stableUnderBaseChange_mk (hQ : Q.StableUnderBaseChange) :
    P.StableUnderBaseChange :=
```
Maybe `stableUnderBaseChange_of` is a better name?

Reply:
I changed it to `stableUnderBaseChange`.